src/ray/raylet/node_manager.cc

```diff
@@ -1370,22 +1423,29 @@ void NodeManager::FinishAssignedTask(Worker &worker) {
     // If this was an actor creation task, then convert the worker to an actor.
     auto actor_id = task.GetTaskSpecification().ActorCreationId();
     worker.AssignActorId(actor_id);
-    const auto driver_id = task.GetTaskSpecification().DriverId();
-
     // Publish the actor creation event to all other nodes so that methods for
     // the actor will be forwarded directly to this node.
-    auto actor_notification = std::make_shared<ActorTableDataT>();
-    actor_notification->actor_id = actor_id.binary();
-    actor_notification->actor_creation_dummy_object_id =
-        task.GetTaskSpecification().ActorDummyObject().binary();
-    actor_notification->driver_id = driver_id.binary();
-    actor_notification->node_manager_id =
-        gcs_client_->client_table().GetLocalClientId().binary();
-
-    RAY_LOG(DEBUG) << "Publishing actor creation: " << actor_id
-                   << " driver_id: " << driver_id;
-    RAY_CHECK_OK(gcs_client_->actor_table().Append(JobID::nil(), actor_id,
-                                                   actor_notification, nullptr));
+    int remaining_reconstructions;
+    int log_length;
+    const auto actor_entry = actor_registry_.find(actor_id);
+    if (actor_entry == actor_registry_.end()) {
+      remaining_reconstructions = task.GetTaskSpecification().MaxActorReconstructions();
+      log_length = 0;
+    } else {
+      // If we've already seen this actor, it means that this actor was reconstructed.
+      // Thus, its previous state must be BEING_RECONSTRUCTED.
+      // Also, we should subtract its remaining_reconstructions by 1.
+      RAY_CHECK(actor_entry->second.GetState() == ActorState::BEING_RECONSTRUCTED);
+      remaining_reconstructions = actor_entry->second.GetRemainingReconstructions() - 1;
+      log_length =
```

Comment:
It looks like you can move the `log_length` assignment outside of the `if-else` block. Please also add a comment explaining how it's computed.